app/scripts/app.js

```diff
@@ -15,6 +15,18 @@ blocChat.config(['$stateProvider', '$locationProvider', function($stateProvider,
     });
 }]);
 
-blocChat.controller('Home.controller', ['$scope', function($scope) {
+blocChat.factory('Room', ['$firebaseArray', function($firebaseArray) {
+    var roomsUrl = 'https://stewart-bloc-chat.firebaseio.com/rooms';
+    var roomsRef = new Firebase(roomsUrl);
```

Comment:
@RyanBalfanz a few notes regarding the example code at the bottom from checkpoint 2: 

- I changed to `$firebaseArray` because `$firebase` is deprecated.
- Because I'm already using `$firebaseArray`, calling `.asArray()` is no longer applicable
- I also don't see the point in using the `all` method to return rooms.

````
  var rooms = $firebase(firebaseRef.child('rooms')).$asArray();;

  return {
    all: rooms
  }
````
Would appreciate specific CR concerning my factory. It works, but I realize injecting `$firebaseArray` into my controller and then assigning `Room` to `$scope` might be weird. Not sure why, though.